build/js/intlTelInput.js

```diff
@@ -358,23 +358,26 @@ https://github.com/Bluefieldscom/intl-tel-input.git
             } else if (!$.fn[pluginName].startedLoadingAutoCountry) {
                 // don't do this twice!
                 $.fn[pluginName].startedLoadingAutoCountry = true;
-                var ipinfoURL = "//ipinfo.io";
-                if (this.options.ipinfoToken) {
-                    ipinfoURL += "?token=" + this.options.ipinfoToken;
+                var self = this;
+                if (typeof this.options.geoipProvider === "function") {
+                    this.options.geoipProvider(function(resp) {
+                        self._autoCountryLoaded(resp);
+                    });
                 }
-                // dont bother with the success function arg - instead use always() as should still set a defaultCountry even if the lookup fails
-                $.get(ipinfoURL, function() {}, "jsonp").always(function(resp) {
-                    $.fn[pluginName].autoCountry = resp && resp.country ? resp.country.toLowerCase() : "";
-                    if ($.cookie) {
-                        $.cookie("itiAutoCountry", $.fn[pluginName].autoCountry, {
-                            path: "/"
-                        });
-                    }
-                    // tell all instances the auto country is ready
-                    // TODO: this should just be the current instances
-                    $(".intl-tel-input input").intlTelInput("autoCountryLoaded");
+            }
+        },
+        _autoCountryLoaded: function(country_code) {
+            $.fn[pluginName].autoCountry = country_code.toLowerCase();
+            if ($.cookie) {
+                $.cookie("itiAutoCountry", $.fn[pluginName].autoCountry, {
+                    path: "/"
                 });
             }
+            // tell all instances the auto country is ready
+            // TODO: this should just be the current instances
+            // $(".intl-tel-input input").intlTelInput("autoCountryLoaded");
+            // Why not just call it directly??
+            this.autoCountryLoaded();
```

Comment:
We cannot just call `autoCountryLoaded` directly, because if there were any other inputs instantiated at the same time as this one, they will also be waiting for this callback (e.g. if you instantiated the plugin on multiple inputs at once - see line 419 where we check `startedLoadingAutoCountry` - if this is set to true, then those instances don't do anything - they are just waiting). Does that make sense?